src/Akka.Persistence.Sql.Linq2Db/Query/Linq2DbReadJournal.cs

```diff
@@ -273,12 +299,98 @@ IEnumerable<string> next(string id)
         private Source<EventEnvelope, NotUsed> _currentEventsByTag(string tag,
             long offset)
         {
-            return Source.FromTask(readJournalDao.MaxJournalSequence())
-                .ConcatMany(
-                    maxInDb =>
+            
+            return AsyncSource<long>.FromEnumerable(new{readJournalDao},async (input)=>
+            {
+                return new[]{ await input.readJournalDao.MaxJournalSequenceAsync()};
+            }).ConcatMany(
+                maxInDb =>
+                {
+                    return _eventsByTag(tag, offset, Some.Create(maxInDb));
+                });
+        }   
+        
+        private Source<EventEnvelope, NotUsed> _events(
```

Comment:
There is probably some opportunity here for re-use. `_events` and `_eventsByTag` are nearly identical.